portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java

```diff
@@ -32,21 +34,35 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     //get user status
     UserService userService = UserServiceFactory.getUserService();
 
+    boolean isLoggedIn;
+    String html;
+    String email;
+
     if (userService.isUserLoggedIn()) {
       //if user is logged in return a greeting and log out button
-      String userEmail = userService.getCurrentUser().getEmail();
+      isLoggedIn = true;
+      email = userService.getCurrentUser().getEmail();
+
       String urlToRedirectToAfterUserLogsOut = "/";
       String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
-
-      response.getWriter().println("<p>Hello " + userEmail + "!</p>");
-      response.getWriter().println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
+      html = "<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>";
     } else {
       //if user is not logged in, return a log in button
+      isLoggedIn = false;
+      email = null;
```

Comment:
this would be the values of these variable at this point of execution. If you like you can move this assignment to the declaration of these variable

Reply:
Good point, done.